engine/common/components-tests/src/test/scala/pl/touk/nussknacker/engine/common/components/DecisionTableSpec.scala

```diff
@@ -0,0 +1,162 @@
+import pl.touk.nussknacker.test.ValidatedValuesDetailedMessage
+
+import java.util.{List => JList, Map => JMap}
+import scala.jdk.CollectionConverters._
+import scala.reflect.ClassTag
+
+trait DecisionTableSpec extends AnyFreeSpec with Matchers with ValidatedValuesDetailedMessage {
+
+  import spel.Implicits._
+
+  "Decision Table component should" - {
+    "filter and return decision table's rows filtered by the expression" in {
+      val scenarioWithDecisionTable = ScenarioBuilder
+        .requestResponse("test scenario")
+        .source("request", TestScenarioRunner.testDataSource)
+        .enricher(
+          "decision-table",
+          "dtResult",
+          "decision-table",
+          "Basic Decision Table" -> decisionTableJson,
+          "Expression"           -> "#ROW['B'] == 'foo' && #ROW['C'] != null",
+        )
+        .end("end", "value" -> "#dtResult")
+
+      val validatedResult = runScenario[TestMessage, JList[JMap[String, Any]]](scenarioWithDecisionTable)(
+        withData = List(TestMessage("1", 100))
+      )
+
+      val resultList = validatedResult.validValue.successes
+      resultList should be(oneElementList {
+        List(
+          Map(
+            "somename" -> 1.0,
+            "B"        -> "foo",
+            "C"        -> "bar"
+          ).asJava
+        ).asJava
+      })
+    }
+  }
+
+  protected def testScenarioRunner: TestScenarioRunner
+
+  protected def runScenario[DATA: ClassTag, RESULT](scenario: CanonicalProcess)(
+      withData: Iterable[DATA]
+  ): ValidatedNel[ProcessCompilationError, RunListResult[RESULT]]
+
+  protected def addEndNode(
+      builder: GraphBuilder[CanonicalProcess]
+  )(id: String, params: Seq[(String, Expression)]): CanonicalProcess
+
+  private lazy val decisionTableJson = Expression.tabularDataDefinition {
+    s"""{
+       |  "columns": [
+       |    {
+       |      "name": "somename",
+       |      "type": "java.lang.Double"
+       |    },
+       |    {
+       |      "name": "B",
+       |      "type": "java.lang.String"
+       |    },
+       |    {
+       |      "name": "C",
+       |      "type": "java.lang.String"
+       |    }
+       |  ],
+       |  "rows": [
+       |    [
+       |      null,
+       |      null,
+       |      "test"
+       |    ],
+       |    [
+       |      "1",
+       |      "foo",
+       |      "bar"
+       |    ],
+       |    [
+       |      null,
+       |      null,
+       |      "xxx"
+       |    ]
+       |  ]
+       |}""".stripMargin
+  }
+
+  private def oneElementList[T](obj: T) = List(obj)
+
+  private implicit class AddEndNodeExt(builder: GraphBuilder[CanonicalProcess]) {
+    def end(id: String, params: (String, Expression)*): CanonicalProcess =
+      addEndNode(builder)(id, params)
+  }
+
+}
+
+private final case class TestMessage(id: String, value: Int)
+
+class FlinkEngineRunDecisionTableSpec extends DecisionTableSpec with FlinkSpec {
```

Comment:
IMO we could have only a lite test, I see no benefits for having tests of components that have the same implementation (using Service) for both engines, but it is a detail - do what you think.

Reply:
I remember you said it's enough to test with the Lite engine. 
But I thought I would check how fast is the Flink-based engine (based on the mini cluster) in tests too. I see it's pretty fast, so the cost of the additional spec is small. Maybe in the future, it will allow us to detect some problems with the production code or at least the test code. 

So, I will leave it as it's.

Reply:
Flink test are quite expensive, but let it be.